shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -56,35 +59,58 @@ class FileChooserDialog {
     else if (action == GTK_FILE_CHOOSER_ACTION_OPEN)
       confirm_text = _("_Open");
 
-    dialog_ = gtk_file_chooser_dialog_new(
-        settings.title.c_str(), NULL, action, _("_Cancel"), GTK_RESPONSE_CANCEL,
-        confirm_text, GTK_RESPONSE_ACCEPT, NULL);
+    gtk_module_ = g_module_open("libgtk-3.so", G_MODULE_BIND_LAZY);
+    void* (*dl_gtk_file_chooser_native_new)(const char*, GtkWindow*,
+                                            GtkFileChooserAction, const char*,
+                                            const char*) = nullptr;
+    bool found = g_module_symbol(
```

Comment:
`gtk_module_` should never be NULL, probably fine to handle that ofc but it should be a loud warning when your app using GTK is loading a non-existing so.

Reply:
I am thinking we could make gtk_symbol_ a static class var, that way the only time we go searching for the lib is on the first invocation thoughts? Then it could be `(gtk_module_ != NULL) || g_module_symbol...`

Reply:
Also, to be clearer about my `gtk_module_ != nullptr` suggestion: the version I was reviewing had the test later in the code logic:

```c++
if (gtk_module_ != nullptr && found ...
```

My suggestion was to move this test to before the `g_module_symbol()` call.

Reply:
@ckerr good point on the release. Static fields for the functions might be a better solution. If none of the fields are NULL, don't do a module search. Let's say I go through the process of dynamically calling all the functions, while saving them as static class functions at the same time, do the dynamically loaded functions reside in memory regardless if we create a GModule object or not? @TingPing

Reply:
It doesn't matter if its never released, the module is loaded as long as the process is a live anyway.

Reply:
@ckerr I will defer to your opinion on any static class variables (function pointers/module).